diff --git a/makefile b/makefile
index 062a73ff55..8192e0c925 100644
--- a/makefile
+++ b/makefile
@@ -153,7 +153,7 @@ $(GENDIR)/hwp_id.html :
 
 .PHONY: check_istep_modules
 check_istep_modules: $(OBJS)
-	listdeps.pl $(IMGDIR)  -v
+	./src/build/tools/listdeps.pl $(IMGDIR)  -v
 
 GENCONFIG_TOOL = src/build/tools/hbGenConfig
 
diff --git a/src/build/linker/makefile b/src/build/linker/makefile
index 523b38586c..b6b423ca4f 100644
--- a/src/build/linker/makefile
+++ b/src/build/linker/makefile
@@ -23,11 +23,11 @@
 #
 # IBM_PROLOG_END_TAG
 ROOTPATH = ../../..
-
+HOST_PREFIX := /usr/bin/
 ifdef HOST_BINUTILS_DIR
-BFD_CFLAGS = -I$(HOST_BINUTILS_DIR)/bfd/ -I$(HOST_BINUTILS_DIR)/include/
-BFD_LDFLAGS = $(HOST_BINUTILS_DIR)/bfd/libbfd.a -ldl \
-	      $(HOST_BINUTILS_DIR)/libiberty/libiberty.a -lz
+BFD_CFLAGS = -I$(HOST_BINUTILS_DIR)/include/
+BFD_LDFLAGS = $(HOST_BINUTILS_DIR)/lib/libbfd.a -ldl \
+               $(HOST_BINUTILS_DIR)/lib/libiberty.a -lz
 else
 BFD_LDFLAGS = -lbfd -liberty -lz
 endif
@@ -39,22 +39,22 @@ include $(ROOTPATH)/config.mk
 
 linker: linker.C
 	$(C2) "    CXX        $(notdir $<)"
-	$(C1)$(CCACHE) $(HOST_PREFIX)g++ -O3 -g linker.C -o linker.o -c \
+	$(C1)$(CCACHE) $(HOST_PREFIX)g++-8 -O3 -g linker.C -o linker.o -c \
 					 $(BFD_CFLAGS)
-	$(C1)$(CCACHE) $(HOST_PREFIX)g++ -O3 -g linker.o -o linker \
+	$(C1)$(CCACHE) $(HOST_PREFIX)g++-8 -O3 -g linker.o -o linker \
 				         $(BFD_LDFLAGS)
 	$(C1)rm linker.o
 
 gensyms: gensyms.C
 	$(C2) "    CXX        $(notdir $<)"
-	$(C1)$(CCACHE) $(HOST_PREFIX)g++ -O3 -g gensyms.C -o gensyms.o -c
-	$(C1)$(CCACHE) $(HOST_PREFIX)g++ -O3 -g gensyms.o -o gensyms \
+	$(C1)$(CCACHE) $(HOST_PREFIX)g++-8 -O3 -g gensyms.C -o gensyms.o -c
+	$(C1)$(CCACHE) $(HOST_PREFIX)g++-8 -O3 -g gensyms.o -o gensyms \
 					 -lpthread
 	$(C1)rm gensyms.o
 
 genlist: genlist.C
 	$(C2) "    CXX        $(notdir $<)"
-	$(C1)$(CCACHE) $(HOST_PREFIX)g++ -O3 -g genlist.C -o genlist.o -c
-	$(C1)$(CCACHE) $(HOST_PREFIX)g++ -O3 -g genlist.o -o genlist \
+	$(C1)$(CCACHE) $(HOST_PREFIX)g++-8 -O3 -g genlist.C -o genlist.o -c
+	$(C1)$(CCACHE) $(HOST_PREFIX)g++-8 -O3 -g genlist.o -o genlist \
 					 -lpthread
 	$(C1)rm genlist.o
diff --git a/src/build/mkrules/cc.rules.mk b/src/build/mkrules/cc.rules.mk
index 8ad26af9f6..83bb9a4d10 100644
--- a/src/build/mkrules/cc.rules.mk
+++ b/src/build/mkrules/cc.rules.mk
@@ -161,8 +161,21 @@ $(OBJDIR)/%.a : $(ARCHIVE_OBJECTS)
 	$(C1)$(if $(STRIP_ARCHIVES), $(STRIP) --strip-debug "$@")
 
 ifdef MODULE
+	$(info MODULE is defined as $(MODULE))
+
+$(info IMGDIR=$(IMGDIR))
+$(info OBJECTS=$(OBJECTS))
+$(info ROOTPATH=$(ROOTPATH))
+$(info MODULE_INIT=$(MODULE_INIT))
+$(info LDFLAGS in mk rule file =$(LDFLAGS))
+
+$(info MODULE is defined...)
 $(IMGDIR)/lib$(MODULE).so : $(OBJECTS) $(ROOTPATH)/src/module.ld $(MODULE_INIT)
 	$(C2) "    LD         $(notdir $@)"
+
+	$(info Linking command: $(C1)$(LD) -shared -z now -x $(LDFLAGS) \
+	$(OBJECTS) $(MODULE_INIT) \
+	-T $(ROOTPATH)/src/module.ld -o $@)
 	$(C1)$(LD) -shared -z now -x $(LDFLAGS) \
 		   $(OBJECTS) $(MODULE_INIT) \
 	           -T $(ROOTPATH)/src/module.ld -o $@
diff --git a/src/build/mkrules/cflags.env.mk b/src/build/mkrules/cflags.env.mk
index 50542dc65e..aa14d65922 100644
--- a/src/build/mkrules/cflags.env.mk
+++ b/src/build/mkrules/cflags.env.mk
@@ -50,7 +50,8 @@ ASMFLAGS += $(COMMONFLAGS) -mcpu=power9 -mbig-endian -ffreestanding -mabi=elfv1
 CXXFLAGS += $(CFLAGS) -nostdinc++ -fno-rtti -fno-exceptions -Werror -Wall \
 	    -fuse-cxa-atexit -std=gnu++17 -Wno-register
 LDFLAGS += --nostdlib --sort-common -EB $(COMMONFLAGS)
-
+CXXFLAGS := $(filter-out -Werror, $(CXXFLAGS))
+CFLAGS := $(filter-out -Werror, $(CFLAGS))
 INCFLAGS = $(addprefix -I, $(INCDIR) )
 ASMINCFLAGS = $(addprefix $(lastword -Wa,-I), $(INCDIR))
 
diff --git a/src/build/mkrules/images.rules.mk b/src/build/mkrules/images.rules.mk
index 584475135d..cd762875df 100644
--- a/src/build/mkrules/images.rules.mk
+++ b/src/build/mkrules/images.rules.mk
@@ -26,7 +26,7 @@
 # File: images.rules.mk
 # Description:
 #     Rules for linking the Hostboot binary images using the custom linker.
-
+LDFLAGS := $(filter-out -fmacro-prefix-map% -fdebug-prefix-map%, $(LDFLAGS))
 ifdef IMGS
 _IMGS = $(addprefix $(IMGDIR)/, $(IMGS))
 IMAGES += $(addsuffix .bin, $(_IMGS)) $(addsuffix .elf, $(_IMGS))
diff --git a/src/build/trace/makefile b/src/build/trace/makefile
index 549dece54a..ceb64363bb 100644
--- a/src/build/trace/makefile
+++ b/src/build/trace/makefile
@@ -28,11 +28,10 @@ SUBDIRS += tracelite.d
 
 GEN_PASS_BODY += tracehash extracthash
 CLEAN_TARGETS += tracehash tracehash.o extracthash extracthash.o
-
 ifdef HOST_BINUTILS_DIR
-BFD_CFLAGS = -I$(HOST_BINUTILS_DIR)/bfd/ -I$(HOST_BINUTILS_DIR)/include/
-BFD_LDFLAGS = $(HOST_BINUTILS_DIR)/bfd/libbfd.a -ldl \
-	      $(HOST_BINUTILS_DIR)/libiberty/libiberty.a -lz
+BFD_CFLAGS = -I$(HOST_BINUTILS_DIR)/include/
+BFD_LDFLAGS = $(HOST_BINUTILS_DIR)/lib/libbfd.a -ldl \
+	       $(HOST_BINUTILS_DIR)/lib/libiberty.a -lz
 else
 BFD_LDFLAGS = -lbfd
 endif
diff --git a/src/usr/diag/prdf/rule/makefile b/src/usr/diag/prdf/rule/makefile
index cf2fda6d5d..c1f6000294 100755
--- a/src/usr/diag/prdf/rule/makefile
+++ b/src/usr/diag/prdf/rule/makefile
@@ -153,7 +153,7 @@ ${OBJ_PLUG_DIR}:
 ${PRDR_CMP_YACC_C_PATH} \
 ${PRDR_CMP_YACC_H_PATH}: ${PRDR_CMP_YACC} | ${OBJ_RULE_DIR}
 	${PRD_OUT_MAKE}
-	$(C1)bison -d -o ${PRDR_CMP_YACC_C_PATH} $^
+	/usr/bin/bison -d -o ${PRDR_CMP_YACC_C_PATH} $^
 
 PRDR_CMP_FLEX: ; # Discard implicit rule for %.lex <- %.lex.C
 ${PRDR_CMP_FLEX_C_PATH}: ${PRDR_CMP_FLEX} | ${OBJ_RULE_DIR}
diff --git a/src/usr/pldm/extended/test/makefile b/src/usr/pldm/extended/test/makefile
index efb63ea8ae..5a1c8b2487 100644
--- a/src/usr/pldm/extended/test/makefile
+++ b/src/usr/pldm/extended/test/makefile
@@ -32,6 +32,6 @@ MODULE = testpldm_extended
 EXTRAINCDIR += ${ROOTPATH}/src/import/
 EXTRAINCDIR += ${ROOTPATH}/src/import/openbmc/pldm/libpldm/include/libpldm
 
-TESTS = *.H
+#TESTS = *.H
 
 include ${ROOTPATH}/config.mk
